Source/LuaBridge/RefCountedPtr.h

```diff
@@ -120,9 +115,41 @@ class RefCountedPtr : private detail::RefCountedPtrBase
     template<typename U>
     RefCountedPtr<T>& operator=(RefCountedPtr<U> const& rhs)
     {
-        reset();
-        m_p = static_cast<T*>(rhs.get());
-        ++getRefCounts()[m_p];
+        // NOTE Self assignment is handled gracefully
+        *this = rhs.get();
+        return *this;
+    }
+
+    RefCountedPtr<T>& operator=(T* const p)
+    {
+        if (p != m_p)
+        {
+            // First increment the counter of the new object to avoid issues with nested objects
+            if (p)
+            {
+                // NOTE Insertion takes only place if not already existing
+                const RefCountsType::iterator itCounterNew =
+                    getRefCounts().insert(std::make_pair(p, 0)).first;
+                ++itCounterNew->second;
+            }
+
+            // Decrease counter of the previous object
+            if (m_p)
+            {
+                const RefCountsType::iterator itCounterPrevious = getRefCounts().find(m_p);
+                assert(itCounterPrevious != getRefCounts().end());
+
+                // Delete object and counter if this was the last user
+                if (--itCounterPrevious->second <= 0)
+                {
+                    delete m_p;
+                    getRefCounts().erase(itCounterPrevious);
```

Comment:
Good finding! Obsolete entries are not removed in the existing implementation.

Reply:
This is the finding which was addressed by @rpatters1 with #268.